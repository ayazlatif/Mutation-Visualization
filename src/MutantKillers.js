import React from 'react';
import './styles/App.css';
import './styles/MutantDisplay.css';

function MutantKillers(props) {
    if (props.killers.length === 0) return null;

    const tableRows = props.killers.map(function (killer, i) {
        return (
            <li key={i}>
                <strong>{killer[0]}</strong>
                <ul>
                    <li>{killer[1]}</li>
                </ul>
            </li>
        );
    });

    return (
        <div className="topLevel">
            <h4>This mutant was killed by the following tests:</h4>
            <ul>
                {tableRows}
            </ul>
        </div>
    );
};

export default MutantKillers;