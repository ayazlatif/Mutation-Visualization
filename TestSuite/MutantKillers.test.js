import React from 'react';
import { shallow, mount } from 'enzyme';

import renderer from 'react-test-renderer';
// Components
import MutantKillers from '../src/MutantKillers';
import { wrap } from 'module';


describe('MutantKillers Test Suite', () => {
    const props = {
        killers: [
            [
                "test_add_10_2 (sample_test.SampleTest)",
                "Traceback (most recent call last):\n  File \"../../example/sample_test.py\", line 9, in test_add_10_2\n    self.assertEqual(11, sample.add10(1))\nAssertionError: 11 != 10\n"
            ],
            [
                "test_add_10 (sample_test.SampleTest)",
                "Traceback (most recent call last):\n  File \"../../example/sample_test.py\", line 6, in test_add_10\n    self.assertEqual(20, sample.add10(10))\nAssertionError: 20 != 100\n"
            ]
        ]
    }
    const wrapper = shallow(<MutantKillers {...props} />);
    const mountWrapper = mount(<MutantKillers {...props} />);

    it('Matches snapshot', () => {
        const tree = renderer.create(<MutantKillers {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Component renders without crashing providing proper JSON', () => {
        expect(wrapper.exists()).toBe(true);
        expect(mountWrapper.exists()).toBe(true);
    });

    it('Displays killed tests correctly', () => {
        expect(wrapper.find('li')).toHaveLength(4);
        expect(wrapper.find('strong')).toHaveLength(2);
        expect(wrapper.find('ul')).toHaveLength(3);
        expect(wrapper.find('h4')).toHaveLength(1);
    });

    it('Returns null if no killers', () => {
        const emptyProps = {
            killers: []
        }
        let result = MutantKillers(emptyProps)
        expect(result).toBe(null);
    });

});