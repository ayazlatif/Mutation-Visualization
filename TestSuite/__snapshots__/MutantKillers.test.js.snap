// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MutantKillers Test Suite Matches snapshot 1`] = `
<div>
  <h4>
    This mutant was killed by the following tests:
  </h4>
  <ul>
    <li>
      <strong>
        test_add_10_2 (sample_test.SampleTest)
      </strong>
      <ul>
        <li>
          Traceback (most recent call last):
  File "../../example/sample_test.py", line 9, in test_add_10_2
    self.assertEqual(11, sample.add10(1))
AssertionError: 11 != 10

        </li>
      </ul>
    </li>
    <li>
      <strong>
        test_add_10 (sample_test.SampleTest)
      </strong>
      <ul>
        <li>
          Traceback (most recent call last):
  File "../../example/sample_test.py", line 6, in test_add_10
    self.assertEqual(20, sample.add10(10))
AssertionError: 20 != 100

        </li>
      </ul>
    </li>
  </ul>
</div>
`;
